OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -118,20 +119,24 @@ public Harvester(Actor self, HarvesterInfo info)
 
 		void INotifyCreated.Created(Actor self)
 		{
+			notify = self.TraitsImplementing<INotifyHarvesterAction>().ToArray();
+
 			if (Info.SearchOnCreation)
 				self.QueueActivity(new FindResources(self));
 		}
 
 		void INotifyBuildComplete.BuildingComplete(Actor self)
 		{
+			notify = self.TraitsImplementing<INotifyHarvesterAction>().ToArray();
```

Comment:
I remember now.
The 'move out of factory' overrides the FindResources queued by Created, that's what this was working around.